src/diabetes/data_processor.py

```diff
@@ -45,5 +48,33 @@ def preprocess_data(self):
             ]
         )
 
-    def split_data(self, test_size, random_state):
-        return train_test_split(self.X, self.y, test_size=test_size, random_state=random_state)
+        self.X_transformed = self.preprocessor.fit_transform(self.X)
+        self.X_df_transformed = pd.DataFrame(self.X_transformed, columns=self.preprocessor.get_feature_names_out())
+
+        return self.X_df_transformed
+
+    def split_data(self, test_size=None, random_state=None):
+        if test_size is None:
+            test_size = self.config["test_size"]
+
+        if random_state is None:
+            random_state = self.config["seed"]
+
+        return train_test_split(self.X_df_transformed, self.y, test_size=test_size, random_state=random_state)
+
+    def pandas_df_to_delta(self, df, name, spark):
+        self._pandas_to_spark_to_delta_cdf(df, name, spark)
+
+    def save_raw_data_to_catalog(self, spark: SparkSession):
+        self._pandas_to_spark_to_delta_cdf(self.df, "raw_data", spark)
+
+    def _pandas_to_spark_to_delta_cdf(self, pandas_df: pd.DataFrame, tbl_name: str, spark: SparkSession):
+        spark_df = spark.createDataFrame(pandas_df).withColumn(
+            "update_timestamp_utc", to_utc_timestamp(current_timestamp(), "UTC")
+        )
+
+        delta_table_path = f"{self.config['catalog_name']}.{self.config['schema_name']}.{tbl_name}"
+
+        spark_df.write.mode("overwrite").saveAsTable(delta_table_path)
+
+        spark.sql(f"ALTER TABLE {delta_table_path} " "SET TBLPROPERTIES (delta.enableChangeDataFeed = true);")
```

Comment:
you do not need  " "before set table properties - just being picky here, no big deal.